earn/src/pages/InfoPage.tsx

```diff
@@ -215,9 +216,29 @@ export default function InfoPage() {
         }
       });
 
+      // Get the time at which each pool was last updated via the oracle (using the update event and getLogs)
+      const latestTimestamps = await Promise.all(
+        poolAddresses.map(async (addr) => {
```

Comment:
Since you're iterating over all poolAddresses anyway, you could just _not_ filter logs by the address. The only downside with that is that if one of the pools hasn't been updated in a while, it may be so old as to not appear in the list.

Does alchemy return an unlimited number of logs? Or is this paginated somehow. If it's unlimited, then I'd definitely remove the filter and just fetch all the logs once --> filter locally.